### PR TITLE
Write test for plotting function plot_carbon_bar

### LINKED ISSUE #370: Write test for plot_carbon_bar
- [x] Check scenarios are in analyze state
- [x] Write tests for the function

### PR DESCRIPTION
**Pull request by rouille**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Write test for plot_carbon_bar plotting function. Closes #370.

### What the code is doing
Add a check to ensure that scenarios are in analyze state and use `plot_show` parameter across the entire function (a `if` statement was on `show_plot`) 

### Testing
New unit tests

### Where to look
New test module and small changes in plotting function

### Usage Example/Visuals
N/A

### Time estimate
5min

### CODE REVIEW

File: postreise/plot/tests/test_plot_carbon_bar.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        "h": [61, 63, 65, 67],
+    },
+    index=pd.date_range(start="2016-01-01", periods=4, freq="H"),
+)
+
+s2 = MockScenario({"plant": mock_plant_s2}, pg=mock_pg_s2)
+s2.info["name"] = "Carbon Scenario 2"
+
+
+@patch("postreise.plot.plot_bar_shortfall.Scenario", return_value=[s1, s2])
```
Comment (BainanXia) [inline postreise/plot/tests/test_plot_carbon_bar.py:57]: Given we are using mocked scenario `s1, s2` directly as input variables to the plot function, I guess we don't need to patch the `Scenario` class within the module here.

Comment (rouille) [reply postreise/plot/tests/test_plot_carbon_bar.py]: Indeed. Done

**Review by BainanXia — APPROVED** [review]
Looks good. Thanks!

File: postreise/plot/plot_carbon_bar.py [hunk 2]
```diff
@@ -8,13 +9,13 @@
...
 
-def plot_carbon_bar(*args, labels=None, labels_size=15, show_plot=True):
+def plot_carbon_bar(*args, labels=None, labels_size=15, plot_show=True):
     """Make bar chart of carbon emissions by fuel type for n scenarios.
 
     :param powersimdata.scenario.scenario.Scenario args: scenario instances.
     :param list/tuple/set labels: labels on plot. Default to scenario id.
     :param int labels_size: size of labels.
-    :param bool show_plot: whether to save the plot.
+    :param bool plot_show: whether to save the plot.
```
Comment (jenhagg) [inline postreise/plot/plot_carbon_bar.py:18]: save -> show?

Comment (rouille) [reply postreise/plot/plot_carbon_bar.py]: Done

**Review by jenhagg — APPROVED** [review]

**Pull request merged**